### ISSUE TITLE
Backwards compatibility issue ? no matching function for call to 'VariantRef::set(const char&)'

### ISSUE THREAD

**Issue by vortex314** [bblanchon/ArduinoJson#1513]
```
void commandArguments(JsonObject config, int argc, char **argv) {
  int opt;

  while ((opt = getopt(argc, argv, "f:m:l:v:")) != -1) {
    switch (opt) {
      case 'm':
        config["mqtt"]["connection"] = optarg;
        break;
      case 'f':
        config["configFile"] = optarg;
        break;
      case 'v': {
        char logLevel = optarg[0];
        config["log"]["level"] = logLevel;
        break;
      }
      case 'l':
        config["log"]["file"] = optarg;
```
../ArduinoJson/src/ArduinoJson/Object/MemberProxy.hpp: In instantiation of ‘typename ArduinoJson6173_71::enable_if<(! ArduinoJson6173_71::is_array<TValue>::value), ArduinoJson6173_71::MemberProxy<TParent, TStringRef>&>::type ArduinoJson6173_71::MemberProxy<TParent, TStringRef>::operator=(const TValue&) [with TValue = char; TObject = ArduinoJson6173_71::MemberProxy<ArduinoJson6173_71::ObjectRef, const char*>; TStringRef = const char*; typename ArduinoJson6173_71::enable_if<(! ArduinoJson6173_71::is_array<TValue>::value), ArduinoJson6173_71::MemberProxy<TParent, TStringRef>&>::type = ArduinoJson6173_71::MemberProxy<ArduinoJson6173_71::MemberProxy<ArduinoJson6173_71::ObjectRef, const char*>, const char*>&]’:
main.cpp:114:34:   required from here
../ArduinoJson/src/ArduinoJson/Object/MemberProxy.hpp:63:5: error: no matching function for call to ‘ArduinoJson6173_71::VariantRef::set(const char&)’
   63 |     getOrAddUpstreamMember().set(src);
      |     ^~~~~~~~~~~~~~~~~~~~~~
The offending line is : 
```
        config["log"]["level"] = logLevel;
```

**Comment by vortex314** [comment]
Adapted the code and works again : 
```
        std::string logLevel;
        logLevel += optarg[0];
        config["log"]["level"] = logLevel;
```
Looks like a char assignment fails, could be JSON-wise indeed unacceptable.

**Comment by bblanchon** [comment]
Hi @vortex314,

Assuming you're talking about the master branch, then yes: the `char` type is not supported anymore.
This change was needed to fix #1498.
As explained in the [changelog](https://github.com/bblanchon/ArduinoJson/blob/54d4b308f4e839c7986c913f77b77be96d9b03fb/CHANGELOG.md), you can use an `int8_t` instead.

Best Regards,
[Benoit](https://paypal.me/benoitblanchon)

**Comment by vortex314** [comment]
Ok, adapted my code. Thanks for the follow-up.
